pennylane/interfaces/autograd.py

```diff
@@ -511,19 +519,29 @@ def grad_fn(dy):
             # TODO: remove this exceptional case once the source of this issue
             # https://github.com/PennyLaneAI/pennylane-sf/issues/89 is determined
             return (return_vjps,)  # pragma: no cover
+
         return return_vjps
 
     return grad_fn
 
 
-def _compute_vjps_autograd(jacs, dy, multi_measurements):
+def _compute_vjps_autograd(jacs, dy, multi_measurements, shots):
     """Compute the vjps of multiple tapes, directly for a Jacobian and co-tangents dys."""
     vjps = []
     for i, multi in enumerate(multi_measurements):
-        if multi:
-            vjps.append(qml.gradients.compute_vjp_multi_new(dy[i], jacs[i]))
-        else:
-            vjps.append(qml.gradients.compute_vjp_single_new(dy[i], jacs[i]))
+        shot_vector = isinstance(shots, Sequence)
+        dy_ = dy[i] if shot_vector else (dy[i],)
+        jac_ = jacs[i] if shot_vector else (jacs[i],)
+
+        shot_vjps = []
+        for d, j in zip(dy_, jac_):
+            if multi:
+                shot_vjps.append(qml.gradients.compute_vjp_multi_new(d, j))
+            else:
+                shot_vjps.append(qml.gradients.compute_vjp_single_new(d, j))
+
+        vjps.append(qml.math.sum(qml.math.stack(shot_vjps), 0))

```

Comment:
Why do we have `0` in the sum here? Is it an implicit dtype casting?

Reply:
It's the axis to sum over (in this case, the shot vector axis). I added to keyword arg to make this explicit.